src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,213 @@
+        val api = SourceApi(readTextResource("/authenticationTest/$testPath"))
+        return MicronautControllerInterfaceGenerator(Packages(basePackage), api).generate().files
+    }
+
+    @BeforeEach
+    fun init() {
+        MutableSettings.updateSettings(
+            genTypes = setOf(CodeGenerationType.CONTROLLERS),
+            controllerTarget = ControllerCodeGenTargetType.MICRONAUT
+        )
+    }
+
+
+    // global authentication tests
+    @ParameterizedTest
+    @MethodSource("testCases")
+    fun `ensure that global authentication set to any authentication will add the required parameter`(testCase: Pair<String, String>) {
+        val controller = setupTest(testCase.first)
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{ it.toString() == testCase.second}
+        }
+
+        if(testCase.second != "SecurityRule.IS_ANONYMOUS") {
+            assertThat(functionParameter).anySatisfy{ parameter ->
+                assertThat(parameter.name).isEqualTo("authentication")
+            }
+        } else {
+            assertThat(functionParameter).noneSatisfy{ parameter ->
+                assertThat(parameter.name).isEqualTo("authentication")
+            }
+        }
+
+    }
+
+
+    @Test
+    fun `ensure that no authentication defined will add no parameter`() {
+        val controller = setupTest("global_authentication_none.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).noneSatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+            assertThat(annotation.members).anyMatch{ it.toString() == "SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).noneSatisfy{ parameter ->
+            assertThat(parameter.name).isEqualTo("authentication")
+        }
+    }
+
+    // operation level authentication tests
+    @Test
+    fun `ensure that both global security and operation-level security being defined will result in the correctly generated code`() {
+
+        val controller = setupTest("operation_authentication_with_global.yml")
```

Comment:
**nitpick:** `controllers` (the same everywhere else)

Reply:
indeed